### DISCUSSION TITLE
```typeof(Member)```

### DISCUSSION THREAD

**Discussion by TonyValenti** [dotnet/csharplang#8484] [Language Ideas]
Many attributes accept a type parameter.
Today, you have to specify a type directly.
It would be really nice if you could specify a member name as well.
For example:
```
    [KnownType(typeof(Value))]
    public record Parent {
        public Child? Value { get; init; }
    }

    public record Child {

    }

```

**Comment by BreyerW** [comment]
I dont think this is feasible w/o new syntax as this can be ambigious with class named the same as member. Cant dot off of containing class either since nested classes are allowed in c# too

**Reply by Joe4evr** [reply]
But it's illegal for a class to have a member with the same name as the class itself, no?

**Reply by ufcpp** [reply]
```cs
class A
{
    public DateOnly Date { get; set; }

    public Type T => typeof(Date); // Date? DateOnly?
}

struct Date
{
    public int Day;
    public int Month;
    public int Year;
}
```

**Comment by HaloFour** [comment]
Overloads would complicate attempting to get the type of any member method group.
